litebird_sim/coordinates.py

```diff
@@ -1,9 +1,115 @@
 # -*- encoding: utf-8 -*-
+import numpy as np
 
 from astropy.coordinates import BarycentricMeanEcliptic
+from astropy.coordinates import SkyCoord
 
 """The coordinate system used by the framework"""
 DEFAULT_COORDINATE_SYSTEM = BarycentricMeanEcliptic()
 
 """The time scale used by the framework"""
 DEFAULT_TIME_SCALE = "tdb"
+
+e2g = (
+    SkyCoord(
+        x=[1.0, 0.0, 0.0],
+        y=[0.0, 1.0, 0.0],
+        z=[0.0, 0.0, 1.0],
+        frame=DEFAULT_COORDINATE_SYSTEM.name,
+        representation_type="cartesian",
+    )
+    .transform_to("galactic")
+    .data.to_cartesian()
+    .get_xyz()
+    .value
+)
+
+
+def ang2vec(theta, phi):
+    """Transform a direction theta,phi to a unit vector.
+
+    Parameters
+    ----------
+    theta : float, scalar or array-like
+      The angle theta (scalar or shape (N,))
+    phi : float, scalar or array-like
+      The angle phi (scalar or shape (N,)).
+
+    Returns
+    -------
+    vec : array
+      The vector(s) corresponding to given angles, shape is (3,) or (3, N).
+
+    See Also
+    --------
+    https://github.com/healpy/healpy/blob/main/healpy/rotator.py#L657
+    """
+    ct, st, cp, sp = np.cos(theta), np.sin(theta), np.cos(phi), np.sin(phi)
+    vec = np.empty((3, ct.size), np.float64)
+    vec[0, :] = st * cp
+    vec[1, :] = st * sp
+    vec[2, :] = ct
+    return vec.squeeze()
```

Comment:
This looks equivalent to [`healpy.ang2vec`](https://healpy.readthedocs.io/en/latest/generated/healpy.pixelfunc.ang2vec.html) and [`ducc0,healpix,ang2vec`](https://gitlab.mpcdf.mpg.de/mtr/ducc/-/blob/ducc0/python/test/test_healpix.py#L102-L106) (but the latter has a slightly different syntax, as it requires a N×2 matrix instead of two vectors). Why was it redefined here?

Reply:
It is. I just wanted to have a self-consistent module, but if you prefer I can use the healpy function. I must admit that I didn't check ducc0. Btw, in principle also for `rotate_coordinates_e2g` we could use the functions of healpy.rotator. Again if you think it is better to go for the healpy functions, I can rework the module.

Reply:
`ducc0` is using the diffeent syntax to minimize the calling overhead as much as possible. It' true that this is more restrictive than the `healpy` one, and it's one of the main reasons why `healpy` isn't using `ducc0` internally as a back-end yet.

Reply:
In fact, yesterday I did some tests and found that `ducc0.healpix.ang2vec` is ~10% faster than `healpy.ang2vec`.

Reply:
@paganol  for me it's fine to keep our own implementation in the code. I would like to abandon healpy some time in the future in favor of ducc, so the fewer dependencies we have, the better.